api/handlers.go

```diff
@@ -212,12 +240,50 @@ func (api *API) userGET(w http.ResponseWriter, req *http.Request, _ httprouter.P
 
 // userLimitsGET returns the speed limits which apply to this user.
 func (api *API) userLimitsGET(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
-	u := api.userFromRequest(req)
-	if u == nil || u.QuotaExceeded {
+	t, err := tokenFromRequest(req)
+	if err != nil {
 		api.WriteJSON(w, database.UserLimits[database.TierAnonymous])
 		return
 	}
-	api.WriteJSON(w, database.UserLimits[u.Tier])
+	tk, err := jwt.ValidateToken(t)
+	if err != nil {
+		api.WriteJSON(w, database.UserLimits[database.TierAnonymous])
+		return
+	}
+	s, exists := tk.Get("sub")
+	if !exists {
+		api.staticLogger.Tracef("Token without a sub.")
+		api.WriteJSON(w, database.TierAnonymous)
+		return
+	}
+	sub := s.(string)
+	// If the user is not cached, or they were cached too long ago we'll fetch
+	// their data from the DB.
+	userLimitCacheMu.Lock()
+	defer userLimitCacheMu.Unlock()
+	if c, exists := userLimitCache[sub]; !exists || c.LastUpdate.Sub(time.Now()) > userLimitCacheTTL {
+		// we don't want to hold the lock during a DB operation
+		userLimitCacheMu.Unlock()
```

Comment:
Pls don't unlock a mutex that is set to be unlocked in a `defer`.  That really hurts readability.

In this case it's probably best to just do the old fashioned

```go
userLimitCacheMu.Lock()
c, exists := userLimitCache[sub]
userLimitCacheMu.Unlock()

if !exists || c.LastUpdate.Sub(time.Now()) > userLimitCacheTTL {
```

Although in this case updating the cache already requires enough logic to warrant a refactor into a 
```go
type userLimitCache struct {
  // some map 
  // some mutex
}

func (c *userLimitCache) Get(...) {}
func (c *userLimitCache) Update(...) {}
```